app/services/dataload_jobs_service.rb

```diff
@@ -0,0 +1,13 @@
+# Fetch Dataload Jobs that are ready to schedule and update the status to queued
+class DataloadJobsService
+  # Queries for all dataload_job records prior to the current DateTime with a
+  # status of 'scheduled' and updates the status to 'queued' then returns
+  # all jobs with a current status of 'queued'
+  #
+  # return [ActiveRecord::Relation] the records to be serialized to JSON in the API
+  def self.fetch
+    current_datetime = DateTime.now
+    DataloadJob.where(status: "scheduled", date: ...current_datetime).update_all(status: :queued) # rubocop:disable Rails/SkipsModelValidations
+    DataloadJob.where(status: "queued", date: ...current_datetime)
```

Comment:
It's definitely the former. I believe this should not be a concern. First, it should be relatively infrequent (daily?) and I would expect the chances of getting earlier `queued` tasks in subsequent fetches to be unlikely as they should be updated int `in_progress` once airflow picks them up, but if not, then selecting them again should be inconsequential.

Partly be design as well, if there was a problem with airflow I think we want to make sure any `queued` tasks get queried, because that's an indication that they've been requested by the API but not acted upon in airflow.